### PR TITLE
Update testDesiredBalanceShouldConvergeInABigCluster to include disk usage

### PR DESCRIPTION
**Pull request by idegtiarenko**
Currently testDesiredBalanceShouldConvergeInABigCluster does not include shard size and disk simulation. Adding those to make the test more realistic.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java [hunk 1]
```diff
@@ -683,7 +716,7 @@ public void testDesiredBalanceShouldConvergeInABigCluster() {
                 "Failed to converge desired balance for [{}] nodes and [{}] total shards:\n {}",
                 nodes,
                 totalShards,
-                clusterState.getRoutingNodes()
+                clusterState.getRoutingNodes().toString()
```
Comment (idegtiarenko) [inline server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java:719]: Logger has special handling for iterables and without to string it only outputs toString for every RoutingNode rather then `RoutingNodes#toString()`. The latter contains mode details

Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java:719]: ++ what a mess :)

File: server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java [hunk 2]
```diff
@@ -595,6 +603,7 @@ public void testDesiredBalanceShouldConvergeInABigCluster() {
             var replicas = randomIntBetween(1, nodes - 1);
             totalShards += shards * (replicas + 1);
             var inSyncIds = randomList(shards * (replicas + 1), shards * (replicas + 1), () -> UUIDs.randomBase64UUID(random()));
+            var shardSize = randomLongBetween(1_000L, 1_000_000L);
```
Comment (DaveCTurner) [inline server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java:606]: This is in bytes right? Seems kinda small if so? We should have some shards in the MiBs range (but not kiB really) and many in the 10s-of-GiBs range.

Comment (idegtiarenko) [reply server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputerTests.java]: Correct, I updated that, however I do not think this is very important as node size is determined based on sum of shard sizes.

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**